rstar/src/algorithm/rstar.rs

```diff
@@ -125,24 +128,24 @@ where
     if node.children.len() < expand_index {
         // Force insertion into this node
         node.children.push(t);
-        return resolve_overflow::<_, Params>(node, current_height);
+        return Ok(resolve_overflow::<_, Params>(node, current_height));
     }
 
     let expand = if let RTreeNode::Parent(ref mut follow) = node.children[expand_index] {
```

Comment:
I think this line also good to compare to the panic invocation below. This indexing into `node.children` could panic and would do so for the same reason the explicitly visible panic would be triggered, i.e. because the internal function `choose_subtree` gave an incorrect result.